src/v2/Apps/Order/Components/CreditCardInput.tsx

```diff
@@ -53,12 +52,12 @@ const StyledBorderBox = styled(BorderBox)<BorderProps>`
 `
 
 interface CreditCardInputProps {
-  error?: StripeError
   onChange?: (response: StripeCardNumberElementChangeEvent) => void
 }
 
 export const CreditCardInput: React.FC<CreditCardInputProps> = props => {
   const [focused, setFocused] = useState(false)
+  const { errors, setFieldError } = useFormikContext<BillingInfoWithTerms>()
```

Comment:
This approach is fine but I guess we could export specific typed context hooks from the ContextProvider and use the exported hook from there instead of using `useFormikContext<BillingInfoWithTerms>()`

From the context provider file, you could have
```jsx
const useBillingInfoWithTermsContext = () => {
  const formikContext = useFormikContext<BillingInfoWithTerms>()
  
  return formikContext
}
```
in this way we make sure we use the right hook attached to the FormContext we want, saves from misusage and typing boilerplate.

Reply:
👍  Similar as Laura mentioned in [another comment](https://github.com/artsy/force/pull/9115#discussion_r778233177), the idea is to hopefully standardize and simplify these datatypes.